match/lib/match/runner.rb

```diff
@@ -172,7 +172,7 @@ def fetch_certificate(params: nil, renew_expired_certs: false, specific_cert_typ
       is_cert_renewable = is_authenticated_with_login || is_cert_renewable_via_api
 
       # Validate existing certificate first.
-      if renew_expired_certs && is_cert_renewable && storage_has_certs
+      if renew_expired_certs && is_cert_renewable && storage_has_certs && !params[:readonly]
```

Comment:
Can you remind me why we need this change? 👀  @nekrich

Reply:
@rogerluan well, we shouldn't be deleting things when we are in readonly mode :)